backends/cloud/storage.cpp

```diff
@@ -72,8 +74,10 @@ void Storage::requestFinishedCallback(Networking::Request *invalidRequestPointer
 	--_runningRequestsCount;
 	if (_syncRestartRequestsed)
 		restartSync = true;
-	if (_runningRequestsCount == 0 && !restartSync)
+	if (_runningRequestsCount == 0) {
```

Comment:
restartSync is removed as condition, why?

Reply:
To make sure the `taskStarted` and `taskFinished` calls are balanced.
If `restartSync` is true, this calls `syncSaves()` a few lines below, which in turn will call `addRequest` and then `taskStarted` if `_runningRequestsCount` is true. So if I had kept the old check we would have two consecutive calls to `taskStaretd` with a single `taskFinished` at the end.

I decided that for the debug message it did not really matter, so I change the test for it as well to keep the code simpler.